### PR TITLE
Implement Inliner

### PR DESCRIPTION
**Pull request by joaonmatos**

### CODE REVIEW

File: ClavaLaraApi/src-lara-clava/clava/clava/code/Inliner.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
+laraImport("clava.ClavaJoinPoints");
+
+class Inliner {
+  #variablePrefix;
+  #variableIndex;
+  constructor() {
+    this.#variableIndex = 0;
+    this.#variablePrefix = "__inline";
```
Comment (joaobispo) [inline ClavaLaraApi/src-lara-clava/clava/clava/code/Inliner.js:8]: If we successively inline several calls inside calls, this means that we can get progressively larger variables names, like __inline__inline__inline_a_0_1_2?

Should this be parameterizable, or it is important to control this name?

File: ClavaLaraApi/src-lara-clava/clava/clava/code/Inliner.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  #getInlinedVarName(originalVarName) {
+    return `${this.#variablePrefix}_${this.#variableIndex}_${originalVarName}`;
+  }
+
+  inline($exprStmt) {
```
Comment (joaobispo) [inline ClavaLaraApi/src-lara-clava/clava/clava/code/Inliner.js:15]: If we want to recursively inline calls inside calls inside calls, the intent of this class is to be used as a building block for that?

**Comment by joaonmatos** [comment]
Current dependencies on/for this PR:
* master
  * **PR #73** <a href="https://app.graphite.dev/github/pr/specs-feup/clava/73" target="_blank">Graphite</a> 
    * **PR #72** <a href="https://app.graphite.dev/github/pr/specs-feup/clava/72" target="_blank">Graphite</a> 
      * **PR #76** <a href="https://app.graphite.dev/github/pr/specs-feup/clava/76" target="_blank">Graphite</a> 
        * **PR #74** <a href="https://app.graphite.dev/github/pr/specs-feup/clava/74" target="_blank">Graphite</a> 
          * **PR #75** <a href="https://app.graphite.dev/github/pr/specs-feup/clava/75" target="_blank">Graphite</a> 
            * **PR #78** <a href="https://app.graphite.dev/github/pr/specs-feup/clava/78" target="_blank">Graphite</a> 
              * **PR #79** <a href="https://app.graphite.dev/github/pr/specs-feup/clava/79" target="_blank">Graphite</a> 
                * **PR #80** <a href="https://app.graphite.dev/github/pr/specs-feup/clava/80" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/specs-feup/clava/80?utm_source=stack-comment).

**Comment by joaonmatos** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/specs-feup/clava/80?utm_source=stack-comment).


Job Reference: rvWNJd05cwCkJ6Hklrdh

**Pull request merged**